src/map/utils/petutils.cpp

```diff
@@ -1602,11 +1602,19 @@ namespace petutils
         {
             PPet->SetMLevel(PMaster->GetMLevel());
             PPet->health.maxhp = (uint32)floor((250 * PPet->GetMLevel()) / 15);
-            PPet->health.hp    = PPet->health.maxhp;
+
+            if (PMaster->StatusEffectContainer->HasStatusEffect(EFFECT_BOLSTER))
+            {
+                uint8 bolsterJPVal = dynamic_cast<CCharEntity*>(PMaster)->PJobPoints->GetJobPointValue(JP_BOLSTER_EFFECT);
```

Comment:
No harm here, and no need to change it, but:

`dynamic_cast` is only required over `static_cast` if you're going to check the result afterwards. Otherwise there's no additional benefit.
```cpp
// If thing wasn't valid for casting, PChar will be nullptr
auto* PChar = dynamic_cast<CCharEntity*>(thing);
if (PChar)
{
    // Only possible to come in here if the dynamic_cast was successful
}

...

// static_cast is pretty much stuffing bits into a CCharEntity ptr, so it might "succeed" even if thing wasn't valid
auto* PChar = static_cast<CCharEntity*>(thing);
if (PChar)
{
    // We can sometimes make it in here, even if we aren't supposed to!
}
```

Note that in the static_cast example, we do this when dealing with Allies as Pets, and we really shouldn't be. But the code relies on that so 🤷

Reply:
ah ok ty for the examples